Adyen/Service/TerminalApiAsyncService.cs

```diff
@@ -0,0 +1,124 @@
+        /// </summary>
+        /// <param name="saleToPoiRequest"><see cref="SaleToPOIRequest"/> or <see cref="SaleToPOIMessage"/>.</param>
+        /// <param name="encryptionCredentialDetails"><see cref="EncryptionCredentialDetails"/>. These must match the credentials that you configured in the Customer Area. Make sure your terminal is updated to latest version.</param>
+        /// <returns>Response <see cref="string"/>.</returns>
+        string RequestEncrypted(SaleToPOIMessage saleToPoiRequest, EncryptionCredentialDetails encryptionCredentialDetails);
+        
+        /// <summary>
+        /// Sends a <see cref="SaleToPOIRequest"/> or <see cref="SaleToPOIMessage"/> to the terminal-api `/async` endpoint.
+        /// </summary>
+        /// <param name="saleToPoiRequest"><see cref="SaleToPOIRequest"/> or <see cref="SaleToPOIMessage"/>.</param>
+        /// <param name="cancellationToken"><see cref="CancellationToken"/>.</param>
+        /// <returns>A <see cref="Task{TValue}"/> that represents the <see cref="string"/>.</returns>
+        Task<string> RequestAsync(SaleToPOIMessage saleToPoiRequest, CancellationToken cancellationToken);
+        
+        /// <summary>
+        /// Sends a <see cref="SaleToPOIRequest"/> or <see cref="SaleToPOIMessage"/> to the terminal-api `/async` endpoint.
+        /// </summary>
+        /// <param name="saleToPoiRequest"><see cref="SaleToPOIRequest"/> or <see cref="SaleToPOIMessage"/>.</param>
+        /// <returns>Response <see cref="string"/>.</returns>
+        string Request(SaleToPOIMessage saleToPoiRequest);
+
+        /// <summary>
+        /// Used to decrypt the notification received.
+        /// </summary>
+        /// <param name="notification">Notification.</param>
+        /// <param name="encryptionCredentialDetails"><see cref="EncryptionCredentialDetails"/>.</param>
+        /// <returns>Decrypted notification <see cref="string"/>.</returns>
+        string DecryptNotification(string notification, EncryptionCredentialDetails encryptionCredentialDetails);
+    }
+
+    public class TerminalApiAsyncService : AbstractService, ITerminalApiAsyncService
+    {
+        private readonly TerminalApiAsyncClient _asyncClient;
+        private readonly SaleToPoiMessageSerializer _saleToPoiMessageSerializer;
+        private readonly SaleToPoiMessageSecuredEncryptor _saleToPoiMessageSecuredEncryptor;
+        private readonly SaleToPoiMessageSecuredSerializer _saleToPoiMessageSecuredSerializer;
+        
+        public TerminalApiAsyncService(Client client) : base(client)
+        {
+            _saleToPoiMessageSerializer = new SaleToPoiMessageSerializer();
+            _saleToPoiMessageSecuredEncryptor = new SaleToPoiMessageSecuredEncryptor();
+            _saleToPoiMessageSecuredSerializer = new SaleToPoiMessageSecuredSerializer();
+            _asyncClient = new TerminalApiAsyncClient(this);
```

Comment:
The merchant will still have to setup a certificate here for the unencrypted flow. The `TerminalLocalApiUnencrypted ` bypasses the `ServerCertificateCustomValidationCallback` of the httphandler to not have to do this. Makes testing a lot easier. Maybe nice to also implement that here (for the unencrypted methods)

Reply:
I like the idea, however, after digging deeper, I noticed it can cause unintended behavior, see example below:

For unencrypted message sending, developers have to override the behavior manually on `HttpClient`-level. The .NET library currently creates a [new Client](https://github.com/Adyen/adyen-dotnet-api-library/blob/v20.0.0/Adyen/Service/TerminalLocalApiUnencrypted.cs#L27). This means that it will override any client settings (timeouts, other settings, baseUrls). I prefer the approach of allowing developers to pass their own HttpClient and be in control of the parameters they pass.

Reply:
Seealso sonar/code scanning: https://github.com/Adyen/adyen-dotnet-api-library/security/code-scanning/4

Reply:
Yeah the sonar scan is a standard message, certificates normally should be verified.

Creating a client, and overriding the existing one is not ideal, however this is not production code. We have got a lot of questions in the past about setting up encryption and certificates hence wanting to make the actual testing part as easy and gradual as possible. You can try to reuse the existing client and override the handler, however I don't think that's necessary as its test code (and cannot make unencrypted calls on live anyways).

Reply:
I've added documentation in the method `<summary>` to showcase how developers can override the `HttpClientHandler` behavior and pass it to the client

Reply:
That works as well